File: week_1/day-5-string-hashing/valid_anagram_follow_up_questions/streamed_input.py
from collections import defaultdict

freq = defaultdict(int)

stream_s = "abc"
stream_t = "bca"

for ch in stream_s:
    freq[ch] += 1

for ch in stream_t:
    freq[ch] -= 1

print(all(v == 0 for v in freq.values()))  # True

File: week_1/day-5-string-hashing/valid_anagram_follow_up_questions/streamed_input.js
const freq = new Map();

const stream_s = "abc";
const stream_t = "bca";

// Process first stream
for (const ch of stream_s) {
    freq.set(ch, (freq.get(ch) || 0) + 1);
}

// Process second stream
for (const ch of stream_t) {
    freq.set(ch, (freq.get(ch) || 0) - 1);
}

// Check if all counts are zero
let isAnagram = true;
for (const value of freq.values()) {
    if (value !== 0) {
        isAnagram = false;
        break;
    }
}

console.log("Are the strings anagrams?", isAnagram);
